### PR TITLE
i.modis.download: more reasonable error handling

### PR DESCRIPTION
**Pull request by neteler**
- catch more (user) errors
- more messages improved
- user example added with .netrc file
- fix broken URL to i.modis.qc
- remove obsolete GRASS GIS version test (hence, it also works with GRASS GIS 8)

### CODE REVIEW

File: grass7/imagery/i.modis/i.modis.download/i.modis.download.py [hunk 1]
```diff
@@ -195,6 +199,8 @@ def main():
                           "the username and password"))
     # set username, password and folder by file
     else:
+        if os.path.isdir(options['settings']):
```
Comment (veroandreo) [inline grass7/imagery/i.modis/i.modis.download/i.modis.download.py:202]: I'm not an expert, but I wonder if `isdir()` will indeed test if a path is a file? I also found that there's `isfile()` (https://docs.python.org/3/library/os.path.html). I think isdir() will always be false, either the file in the path exists or not, at least that's what my toy tests show here.

File: grass7/imagery/i.modis/i.modis.download/i.modis.download.py [hunk 2]
```diff
@@ -255,7 +261,7 @@ def main():
         modisOgg.connect()
         if modisOgg.nconnection <= 20:
             # download tha tiles
-            grass.message(_("Downloading MODIS product <%s>..." % produ))
+            grass.message(_("Downloading MODIS product <%s> (%s)..." % (produ, prod['prod'])))
```
Comment (veroandreo) [inline grass7/imagery/i.modis/i.modis.download/i.modis.download.py:264]: general question: what is the difference between `%s % ()` and the .format() way?

Comment (lucadelu) [reply grass7/imagery/i.modis/i.modis.download/i.modis.download.py:264]: I think format should be preferable. Reading [this](https://stackoverflow.com/questions/5082452/string-formatting-vs-format-vs-f-string-literal) format seems slower...

File: grass7/imagery/i.modis/i.modis.download/i.modis.download.py (hunk 1 above)
Comment (lucadelu) [reply grass7/imagery/i.modis/i.modis.download/i.modis.download.py:202]: I agree

Comment (anikaweinmann) [inline grass7/imagery/i.modis/i.modis.download/i.modis.download.py:202]:
```suggestion
        if not os.path.isfile(options['settings']):
```

Comment (lucadelu) [reply grass7/imagery/i.modis/i.modis.download/i.modis.download.py]: I committed the code but I don't see it in the code....

File: grass7/imagery/i.modis/i.modis.download/i.modis.download.py (hunk 2 above)
Comment (anikaweinmann) [inline grass7/imagery/i.modis/i.modis.download/i.modis.download.py:264]:
```suggestion
            grass.message(_("Downloading MODIS product <{}> ({})...".format(produ, prod['prod'])))
```

Comment (neteler) [reply grass7/imagery/i.modis/i.modis.download/i.modis.download.py:264]: ok, changed by Anika :)

File: grass7/imagery/i.modis/i.modis.download/i.modis.download.py (hunk 1 above)
Comment (neteler) [reply grass7/imagery/i.modis/i.modis.download/i.modis.download.py]: Fixed by Anika

**Review by veroandreo — APPROVED** [review]

**Pull request merged**